src/py/flwr/server/compat/driver_client_proxy_test.py

```diff
@@ -76,13 +77,27 @@ def _make_task(
     )
 
 
+def validate_task_res(
+    task_res: task_pb2.TaskRes,  # pylint: disable=E1101
+) -> None:
+    """Validate if a TaskRes is empty or not."""
+    if not task_res.HasField("task"):
+        raise ValueError("Invalid TaskRes, field `task` missing")
+    if task_res.task.HasField("error"):
+        raise ValueError("Exception during client-side task execution")
+    if not task_res.task.HasField("recordset"):
+        raise ValueError("Invalid TaskRes, both `recordset` and `error` are missing")
+
+
 class DriverClientProxyTestCase(unittest.TestCase):
     """Tests for DriverClientProxy."""
 
     def setUp(self) -> None:
         """Set up mocks for tests."""
-        self.driver = MagicMock()
-        self.driver.get_nodes.return_value = (
+        self.driver = GrpcDriver()
+        self.driver.driver_helper = MagicMock()
```

Comment:
Wouldn't it be better to remove everything that's gRPC-related? Would could use just a mock `Driver` b/c that's what `DriverClientProxy` relies on